enhancements/sig-architecture/64-placementStrategy/README.md

```diff
@@ -0,0 +1,360 @@
+
+```go
+type PlacementSpec struct {
+	// ClusterSets represent the ManagedClusterSets from which the ManagedClusters are selected.
+	// +optional
+	ClusterSets []string `json:"clusterSets,omitempty"`
+
+	// Predicates represent a slice of predicates to select ManagedClusters. The predicates are ORed.
+	// +optional
+	Predicates []ClusterPredicate `json:"predicates,omitempty"`
+            ...
+ 
+	// PlacementStrategy define affinity clusters as subset of the selected clusters and divide the selected clusters to groups.
+	// +optional
+	DecisionGenerator DecisionGenerator `json:"decisionGenerator,omitempty"`
+}
+
+type AffinityClusterGroup struct {
+	// LabelSelector represents a selector of ManagedClusters by label
+	// +optional
+	ClusterSelector metav1.LabelSelector `json:"clusterSelector,omitempty"`
+
+	// ClusterGroupPercentage is a percentage of the total selected clusters that will divide the clusters to groups.
+	// ex; for a total 100 clusters selected, ClusterGroupPercentage equal to 20% will divide the selected clusters to
+	//     5 groups each group should have 20 clusters.
+	// +kubebuilder:validation:Enum="5%";"10%";"15%";"20%";"25%";"30%";"50%";"100%"
+	// +kubebuilder:default:=100%
+	// +optional
+	AffinityGroupPercentage string `json:"affinityGroupPercentage,omitempty"`
+}
+
+type PlacementStrategy struct {
+	// Affinity clusters label selector. Affinity clusters will be placed in cluster group index 0 and it will have MandatoryClusterGroup type.
+	// +optional
+	AffinityClusterGroup AffinityClusterGroup `json:"affinityClusterGroup,omitempty"`
+
+	// ClusterGroupPercentage is a percentage of the total selected clusters that will divide the clusters to groups.
+	// ex; for a total 100 clusters selected, ClusterGroupPercentage equal to 20% will divide the selected clusters to
+	//     5 groups each group should have 20 clusters.
+	// +kubebuilder:validation:Enum=None;5%;10%;15%;20%;25%;30%;50%;100%
+	// +kubebuilder:default:=None
+	// +optional
+	ClusterGroupPercentage string `json:"clustersGroupPercentage,omitempty"`
+
+	// NumberOfClustersPerGroup is the max number of clusters per group. If ClusterGroupPercentage is used the NumberOfClustersPerGroup will be ignored.
+	// +kubebuilder:default:=100
+	// +optional
+	NumberOfClustersPerGroup int32 `json:"numberOfClustersPerGroup,omitempty"` 
+}
+```
+
+Adding ClusterGroupStatus to the PlacementStatus API 
+```go
+type ClusterGroupStatus struct {
```

Comment:
Not so sure why we need this field, all related details of decisions can be got from decisions if self, right? or I think we should have a list of decision info in the placement e.g.

```yaml
status:
  decisions:
  - name: decision-1
    labels:
      purpose: canary
      ...
    numberOfClusters: 10
```

Reply:
```yaml
decisions:
  - names: 
     - decision-1
     - decision-2
    selector:
      matchLabels:
        purpose: canary
    numberOfClusters: 10
```